PURL-TYPES.rst

```diff
@@ -87,6 +87,25 @@ bitnami
       pkg:bitnami/wordpress@6.2.0
       pkg:bitnami/wordpress@6.2.0?arch=arm64
 
+brew
+----
+``brew`` for Homebrew-based packages:
+
+- The default repository (tap) is ``homebrew/core``.
```

Comment:
I think care needs to be taken here to distinguish the default tap repository `homebrew/core` from the PURL `repository_url` qualifier, which takes a URL. `homebrew/core` is not a valid URL and should not be put into the `repository_url` qualifier. Besides being confusing to have something that isn't a URL in the URL qualifier, this would break any implementation that expects the repository URL to always be a URL.

There should be an example that shows specifying the repository.

Reply:
> There should be an example that shows specifying the repository.

Might be misunderstanding what you mean, but examples 2 through 4 currently show this:

* `pkg:brew/homebrew/core/sqlite@3.43.2` specifies the repository as `homebrew/core`, which is otherwise the implicit default repo;
* `pkg:brew/some-org/some-tap/some-app@1.2.3` specifies the repository as `some-org/some-tap`, which expands to `https://github.com/some-org/homebrew-some-tap`, per Homebrew's tap expansion rules;
* `pkg:brew/some-org/some-tap/some-app@1.2.3?tap_url=https://git.example.com/some-org/some-tap.git` specifies the repository as `https://git.example.com/some-org/some-tap.git`, which then gets tapped with a custom name of `some-org/some-tap` (which could be anything, since the full URL is specified)

Reply:
I thought the repository would be in a qualifier like how it is with `repository_url`. If it's in the namespace, that should be called out. Normally, a type spec has a line at the top specifying the default repository URL qualifier value, but `brew` doesn't seem to use the repository URL qualifier. I'm trying to avoid people creating PURLs like `pkg:brew/sqlite@3.43.2?repository_url=homebrew/core` instead.

As it's currently written, there's an explanation of how repositories work, but I don't see an explanation of how the repository is encoded into the PURL. Apparently it goes into the namespace, and there are examples of tap names in the namespace, but the namespace field is never mentioned.

Reply:
> I thought the repository would be in a qualifier like how it is with repository_url. If it's in the namespace, that should be called out.

Gotcha, I'll do this. I'm not super familiar with the idioms here; is the namespace not the right place to put this? A Homebrew Tap is essentially a package namespace, and the purl spec says this about namespaces:

> **namespace**: some name prefix such as a Maven groupid, a Docker image owner,
  a GitHub user or organization. Optional and type-specific.

Reply:
I think namespace should be fine as long as it's explained.

Reply:
Ah, I think I see the ambiguity here: Homebrew's taps don't really distinguish between the "repository" and the "namespace," which are distinct concepts in purls. For standard/consistent language purposes, I'll refer to the tap only as a namespace.